### PR TITLE
Fix RPM packaging

### LINKED ISSUE #269: RPM tells user to start goldstone-server regardless of package installed
Installing goldstone-server-enterprise asks you to start goldstone-server ... which isn't installed.

```
######################################################################## 100.0%
  Installing : 1453679820:goldstone-server-enterprise-0.8.0-SNAPSHOT.188.gbfb3825.rpm.systemd.bugs.el7.x86_64                            1/1
*****************************************************************************

 Modify /goldstone/config/goldstone-prod.env
 before starting goldstone-server. See /goldstone/INSTALL.md
 for details.

 To enable and start goldstone-server, run:

     systemctl enable goldstone-server
     systemctl start goldstone-server

*****************************************************************************
``` 

This should ask you to `systemctl enable goldstone-server-enterprise`

### LINKED ISSUE #384: uninstalling goldstone enterprise RPM does not remove docker images
When I tried to yum uninstall 0.8.3, the docker containers were not shut down, and the images could not be removed.  Uninstallation should ensure that the containers are stopped, and the images are fully removed from the system.

### LINKED ISSUE #385: verify that docker login during enterprise RPM install works
The way it is today, the enterprise systemd startup script does the `docker login` and depends on 3 environment variables to succeed.  

We should either:

a) document the mechanism to set those variables (possibly in the output from yum install) so they are available to systemd
b) move the docker login command to the RPM install (should only need to be done once anyway), and force user to set the env vars for install to succeed
c) document the docker login as a prerequisite to RPM installation

Options b) or c) seem most appropriate to me, but any is acceptable for 1.0.

### LINKED ISSUE #386: should RPM install wait for docker images to pull before completing?
The first time you start goldstone server, it takes several minutes to complete because it is pulling the images from docker repos.  Since it is starting via systemd, any output indicating the status is a bit hard to find.  You may be able to see something in the service status, or by calling `docker images`, but if you didn't know to look for it, you may think that nothing is happening.

We used to pull the images as part of the RPM install, which unfortunately didn't output status until the stdout was dumped at the end of the installation.  The RPM install would take several minutes to complete, but at least you knew that when you started the service, it would happen quickly.

Should we go back to pulling the images during RPM install?

### PR DESCRIPTION
**Pull request by slashk**
fix RPM packaging to stop containers as well as adding vagrant for more complete testing.

closes #269. closes #384. closes #385. closes #386.

### CODE REVIEW

File: desktop [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 01a208d295b862a2f58ef67a6a3a6085b00b4d1b
```
Comment (jxstanford) [inline desktop]: do you have PRs in these other projects or possibly forgot to submodule update?

Comment (slashk) [inline desktop]: yeah, i think i just updated it with the 857af10 commit

File: rpm_packaging/after-install.sh [hunk 2]
```diff
@@ -1,17 +1,39 @@
...
+if [ $PKGNAME="goldstone-server-enterprise" ]; then
+  echo " Before starting $PKGNAME you must set your Goldstone license variables."
+  echo " and login into the custom repository."
+  echo " You can do this via the following commands:        "
+  echo "   # export GOLDSTONE_REPO_USER=<<YOUR_REPO_USER>>  "
+  echo "   # export GOLDSTONE_REPO_PASS=<<YOUR_REPO_PASS>>  "
+  echo "   # export GOLDSTONE_REPO_EMAIL=<<YOUR_REPO_EMAIL>>"
+  echo "   # docker login -u "${GOLDSTONE_REPO_USER}" -p "${GOLDSTONE_REPO_PASS}" -e "${GOLDSTONE_REPO_EMAIL}" gs-docker-ent.bintray.io"
+  echo ""
+  echo "Please contact your Goldstone sales rep is you have any questions."
```
Comment (jxstanford) [inline rpm_packaging/after-install.sh]: "Please contact sales@solinea.com if you have any questions" ?

Comment (slashk) [inline rpm_packaging/after-install.sh]: 👍🏻

**Comment by jxstanford** [comment]
:+1:

**Pull request merged**